### DISCUSSION TITLE
MacBook Pro M3 with Thunderbolt 4 to monitor - charges without restriction

### DISCUSSION THREAD

**Discussion by fearby** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1201] [Q&A]
Am I correct in saying that the Thunderbolt 4 cable used to connect my monitor will charge my MPB and nothing I can do about it (except change to an HDMI cable)?
Thanks

**Comment by MatthiasKerbl** [comment]
Hi @fearby,

I am not sure if I understand your question. Maybe you could elaborate a little bit more.
If your monitor supports passthrough charging, your MacBook will charge, however, if you us AlDente and set a charge limit, your MacBook will only charge to the charge limit.

Best,
Matthias

**Reply by fearby** [reply]
Ok I thought Al dente just limited the charge from the MagSafe connection.

The Al dente capacity seemed to be stuck at 99% even though I set a limit of 70% so assumed it was through the Thunderbolt/USBC connection. I tried using the thunderbolt cable again and battery seems to be dropping appropriately now even though haven't changed anything.

**Reply by MatthiasKerbl** [reply]
No, AlDente will limit charging from any power source.

I am not sure exactly what you mean with "stuck at 99%" but in general, your MacBook will only discharge when it is plugged in when you manually click on Discharge or you have AlDente Pro and you have Automatic Discharge enabled.

Best,
Matthias

**Reply by theomythos** [reply]
Have the same problem. Macbook pro M3 gets the power from an external monitor through thunderbolt, AlDente is set to 80%. If I unplug the MBP and discharge it below 80% then plug back into the monitor, it charges up to 100%.

**Reply by MatthiasKerbl** [reply]
Hi @theomythos,

To assess your issue better, please write our support or create a new issue here on GitHub and fill out the questions for a bug report.

Best,
Matthias
